### PR TITLE
✨amp-delight-player: Additional analytics support

### PR DESCRIPTION
**Pull request by xymw**
- added AD_START and AD_END event support
- added tests for event forwarding
- replace some parts to utilize functions provided by iframe-video module
- implement `seekTo()` interface

### CODE REVIEW

File: extensions/amp-delight-player/0.1/amp-delight-player.js [hunk 1]
```diff
@@ -240,17 +246,28 @@ class AmpDelightPlayer extends AMP.BaseElement {
...
    */
   handleDelightMessage_(event) {
-    if (event.source !== this.iframe_.contentWindow) {
+    if (!originMatches(event, this.iframe_, /.*/)) {
       return;
     }
 
     const data = objOrParseJson(getData(event));
-    if (data === undefined || data['type'] === undefined) {
+    if (data === null || data === undefined || data['type'] === undefined) {
```
Comment (alanorozco) [inline extensions/amp-delight-player/0.1/amp-delight-player.js]: Can we do `if (!data || !data['type'])` instead?

File: extensions/amp-delight-player/0.1/amp-delight-player.js [hunk 2]
```diff
@@ -240,17 +246,28 @@ class AmpDelightPlayer extends AMP.BaseElement {
...
       return; // We only process valid JSON.
     }
 
     const {element} = this;
 
+    redispatch(element, data['type'], {
```
Comment (alanorozco) [inline extensions/amp-delight-player/0.1/amp-delight-player.js]: `redispatch` returns a boolean to short-circuit the message method, can you return?

```js
const redispatched = redispatch(...);
if (redispatch) {
  return;
}
```

**Comment by xymw** [comment]
New analytics related addition:
- Add support for custom events via CUSTOM_TICK

**Comment by xymw** [comment]
We did the changes @alanorozco proposed and awaiting further review or merging of the PR.

**Comment by xymw** [comment]
@alanorozco Any update on this PR? If we can do anything on our end to move forward let me know.

File: extensions/amp-delight-player/0.1/amp-delight-player.js [hunk 3]
```diff
@@ -25,11 +32,12 @@ import {removeElement} from '../../../src/dom';
 import {setStyle} from '../../../src/style';
 import {userAssert} from '../../../src/log';
 
-import {CSS} from '../../../build/amp-delight-player-0.1.css';
-
 /** @const */
 const TAG = 'amp-delight-player';
 
+/** @private @const */
+const ANALYTICS_EVENT_TYPE_PREFIX = 'video-custom-';
```
Comment (alanorozco) [inline extensions/amp-delight-player/0.1/amp-delight-player.js]: Can you add a TODO to export this from a lower level, like `src`? (No need to do for this PR, I'll take care of it)

**Comment by alanorozco** [comment]
@xymw Thank you! Merging for relase.

**Pull request merged**